mentions/views.py

```diff
@@ -7,7 +7,10 @@
 
 
 def autocomplete(request):
-    term = request.GET.get('term', '')
+    # to be haystack+solr friendly we must strip and join spaces
+    term = request.GET.get('term', '').strip()
+    term = ' '.join(term.split())
```

Comment:
That's a good point, I'm just not sure that it is a right place for that piece of code, since not everyone uses solr. This can be done inside of user-defined `Provider.search()`, which makes a search using solr.
At the same time I think this might be useful no matter what search engine you use.